newrelic-agent/src/main/java/com/newrelic/api/agent/NewRelicApiImplementation.java

```diff
@@ -147,6 +163,21 @@ public void noticeError(String message, boolean expected) {
         noticeError(message, params, expected);
     }
 
+    private boolean isExpectedErrorConfigured(String expectedConfigName, String message) {
+        boolean expected = true;
+        String app_name = NewRelic.getAgent().getConfig().getValue("app_name");
+        Set<ExpectedErrorConfig> expectedErrors = ServiceFactory.getConfigService().getErrorCollectorConfig(app_name).getExpectedErrors();
+        for (ExpectedErrorConfig errorConfig : expectedErrors) {
+            if ((errorConfig.getErrorClass().equals(expectedConfigName) && errorConfig.getErrorMessage() == null) ||
+                    (errorConfig.getErrorMessage() != null) &&
+                            (errorConfig.getErrorClass().equals(expectedConfigName) && errorConfig.getErrorMessage().equals(message))) {
```

Comment:
You can store the values from `errorConfig.getErrorClass()` and `errorConfig.getErrorMessage()` in variables before evaluating the if statement to avoid calling the getter methods repeatedly.